### PR TITLE
Early stages of config UI (missing craftables and radio ents)

### PR DESCRIPTION
**Pull request by user-420**
I think we could go ahead and push this so we can get basic GUI based config manipulation into master.

As the title states, craftables and radio ents aren't implemented into the UI yet but I won't be able to do them for some time as I'm gonna be busy fixing stuff on my car for the next while.

Still need a second pair of eyes to go over the code and some others to do some testing of the UI, from what I've tested everything works fine but there's almost always edgecases here and there.

### CODE REVIEW

File: lua/jmod/cl_init.lua [hunk 1]
```diff
@@ -377,10 +377,9 @@ net.Receive("JMod_LuaConfigSync", function(dataLength)
 	JMod.LuaConfig = JMod.LuaConfig or {}
 	JMod.LuaConfig.ArmorOffsets = Payload.ArmorOffsets
 	JMod.Config = JMod.Config or {}
-	JMod.Config.General = JMod.Config.General or {}
-	JMod.Config.Weapons = JMod.Config.Weapons or {}
-	JMod.Config.General.AltFunctionKey = Payload.AltFunctionKey
-	JMod.Config.Weapons.SwayMult = Payload.WeaponSwayMult
+	JMod.Config.General = {["AltFunctionKey"] = Payload.AltFunctionKey}
+	JMod.Config.Machines = {["Blackhole"] = Payload.Blackhole}
+	JMod.Config.Weapons = {["SwayMult"] = Payload.WeaponSwayMult}
```
Comment (AdventureBootCoder) [inline lua/jmod/cl_init.lua:382]: Why were square brackets and quote marks used?
Wouldn't `{AltFunctionKey = PayLoad.AltFunctionKey}` work?
Or is this a workaround to a problem?

**Review by AdventureBootCoder — COMMENTED** [review]
Everything looks good to me, and nothing in `cl_config_ui` struck me as out of place.

File: lua/jmod/cl_init.lua (hunk 1 above)
Comment (user-420) [reply lua/jmod/cl_init.lua:382]: cause its a index of the array, don't you have to use [x] to indicate that?

Comment (user-420) [reply lua/jmod/cl_init.lua:382]: Nvm, ig you don't. I saw us doing that somewhere else so assumed it was necessary.

**Comment by user-420** [comment]
Note: don't delete this branch after the merge, I still intend to use this branch to implement craftables

**Comment by AdventureBootCoder** [comment]
What did Jack make me a lead dev for if not to do stuff while he's away?

**Pull request merged**